packages/teams-js/src/internal/marketplaceUtils.ts

```diff
@@ -0,0 +1,59 @@
+import { marketplace } from '../public';
+
+/**
+ * @hidden
+ * Validates the cart item properties are valid
+ * @param cartItems The cart items
+ * @returns [true, undefined] if all properties of all cart items are valid, [false, error message] otherwise
+ *
+ * @internal
+ * Limited to Microsoft-internal use
+ */
+export function validateCartItems(cartItems: marketplace.CartItem[]): [boolean, string | undefined] {
+  for (const cartItem of cartItems) {
```

Comment:
Other validation I'd like to see in here (or explanations as to why it's not needed):

- id is not null and not empty
- name is not null and not empty
- imageURL, if present, is a valid URL (this is very easy to do in typescript: 
![image](https://github.com/OfficeDev/microsoft-teams-library-js/assets/18446904/cd5a94a6-802a-4880-8ad2-8a8ced72b952))
- accessories, if present, is in a valid state:
     - Array is either absent or contains items
     - Items in the array do not themselves contain nested accessories arrays (accessories can't have more accessories)